test/functional/tests/stress/test_kedr.py

```diff
@@ -0,0 +1,136 @@
+    """
+    title: Loading modules with kedr started with 'memleak' configuration
+    description: Load and unload modules with kedr started to watch for memmory leaks
+    pass_criteria:
+      - No memory leaks observed after loading and unloading module
+    """
+    with TestRun.step(f"Starting kedr against {module}"):
+        Kedr.start(module.value)
+
+    with TestRun.step(f"Loading {module}"):
+        os_utils.load_kernel_module(module.value)
+
+    with TestRun.step(f"Unloading {module}"):
+        os_utils.unload_kernel_module(module.value, os_utils.ModuleRemoveMethod.modprobe)
+
+    with TestRun.step(f"Checking for memory leaks for {module}"):
+        Kedr.check_for_mem_leaks(module.value)
+
+    with TestRun.step(f"Stopping kedr"):
+        Kedr.stop()
+
+
+@pytest.mark.parametrize("module", cas_module.CasModule)
+def test_kedr_fsim_load_cas_module(module, unload_modules, install_kedr):
+    """
+    title: Loading modules with kedr started with 'fsim' configuration
+    description: Load and unload modules with kedr started to simulate kmalloc fails
+    pass_criteria:
+      - Module fails to load
+    """
+    with TestRun.step(f"Starting kedr against {module}"):
+        Kedr.start(module.value, KedrProfile.FAULT_SIM)
+
+    with TestRun.step("Setting up fault simulation parameters"):
+        Kedr.setup_fault_injections()
+
+    with TestRun.step(f"Trying to load {module}"):
+        out = os_utils.load_kernel_module(module.value)
+        if out.exit_code == 0 \
+                or "Cannot allocate memory" not in out.stderr:
+            TestRun.fail(f"Loading module should fail because of alloc error, instead "
+                         f"modprobe's output is: {out}")
+
+    with TestRun.step(f"Stopping kedr"):
+        Kedr.stop()
+
+
+@pytest.mark.parametrize("module", cas_module.CasModule)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_kedr_start_cache(module, unload_modules, install_kedr):
+    """
+    title: Start cache and add core with kedr started against one of CAS modules
+    description: |
+        Load CAS modules, start kedr against one of them, start cache and add core,
+        stop cache and unload modules
+    pass_criteria:
+      - No memory leaks observed
+    """
+    with TestRun.step("Preparing cache device"):
+        cache_device = TestRun.disks['cache']
+        cache_device.create_partitions([Size(500, Unit.MebiByte)])
+        cache_part = cache_device.partitions[0]
+
+    with TestRun.step("Preparing core device"):
+        core_device = TestRun.disks['core']
+        core_device.create_partitions([Size(1, Unit.GibiByte)])
+        core_part = core_device.partitions[0]
+
+    with TestRun.step("Unload CAS modules if needed"):
+        if os_utils.is_kernel_module_loaded(module.value):
+            cas_module.unload_all_cas_modules()
```

Comment:
Isn't it done in fixture (unload_modules) before test?

Reply:
It actually is, but start_cache uses helpers for creating partitions, which I believe in turn calls 'isdct show' command which reloads modules - so after unloading during fixture modules are loaded once more at this point of test...
I could possibly move unloading modules from fixture to beginning of each test but I think it is ok the way it is right now.